### ISSUE TITLE
OverlayWindow without infinite loop in v4

### ISSUE THREAD

**Issue by Hirse** [michel-pi/GameOverlay.Net#21]
With the update to v4 there seem to be two option of creating the overlay:
1. The `OverlayWindow` which requires a `while (true)` loop to continuously render
2. The `GraphicsWindow` which provides a `DrawGraphics` event, but requires a parent window to stick to

In previous versions (3.7.8 for example), I used a `FrameTimer` to set the FPS and use the event for an `OverlayWindow`. Is this option (or something similar) still available in the latest version?

I would like to have an overlay over the whole desktop while being able to run other code.

**Comment by michel-pi** [comment]
I guess it was a pretty bad design choice to make the `GraphicsWindow` stick to a parent window.
The purpose of this one should more likely be the combination of having an `OverlayWindow` and a renderer running.

I was also thinking about getting rid of the automatically stick to a parent window thing.
You can achieve this yourself by using the `FitToWindow` method in the `OverlayWindow`.

But this made me unconfortable in my use case where calling FitToWindow each frame would cause performance problems (maybe) and we would probably run into a problem where moving or resizing is done within `BeginScene` and `EndScene`.

In conclusion:
I would like to change the code to get rid of the sticky window aspect in the GraphicsWindow.
I may also expose a method to attach (join) the window or renderer thread (to wait for it to exit) and create another event which fires before another frame starts.

Any other ideas or problems you see here?

**Comment by Hirse** [comment]
I can see the appeal of having a "StickyOverlayWindow" so you don't have to worry about moving and resizing.

Would it be possible to have a three-tiered inheritance of `OverlayWindow` -> `GraphicsWindow` -> `StickyWindow` where the `GraphicsWindow` adds the renderer and `StickyWindow` adds the dependency on a parent window?

Being not familiar with the graphics and drawing mechanisms (except using this library) I don't really understand your second conclusion.

**Comment by michel-pi** [comment]
Okay i like that idea.
I am going to override https://github.com/michel-pi/GameOverlay.Net/blob/master/source/Windows/GraphicsWindow.cs#L306 in a `StickyWindow` to handle that stuff.

Oh and what i meant in the second part of the conclusion was what i implemented (and forgot about) here:
https://github.com/michel-pi/GameOverlay.Net/blob/master/source/Windows/GraphicsWindow.cs#L175
This Sleep/Wait until the thread stops.

I am also going to add this mechanism to the `OverlayWindow`

I probably add these changes tomorrow if thats okay.

A bit off topic: You used this before i made all those breaking changes. Do you like the new stuff? I tried to be more like a GDI+ (System.Drawing.Graphics) replacement.

**Comment by Hirse** [comment]
Awesome. Thanks for being flexible and open to changes. 👍 

I used the 3.x version, got excited when I saw an update, and watched it break all my stuff. 😄 
I haven't productively used 4.x but it looks like a nicer structure.

**Comment by michel-pi** [comment]
Great. I am going to close this now. Feel free to open more issues if you have any questions, feature requests and so on... :smiley:
